src/core/funcbloc.h

```diff
@@ -164,6 +166,22 @@ class CFunctionBlock : public forte::core::CFBContainer {
       return static_cast<TEventID>(getPortId(paEONameId, getFBInterfaceSpec().mNumEOs, getFBInterfaceSpec().mEONames));
     }
 
+    /*! \brief Gets the EventTypeID of a specific input event
+     * \param paEIID  StringId of the event name.
+     * \return Returns the EventTypeID of a specific input event
+     */
+    CStringDictionary::TStringId getEIType(TEventID paEIID) const {
+      return getFBInterfaceSpec().mEITypeNames[paEIID];
```

Comment:
I suggest returning `g_nStringIdEvent` as default when `mEITypeNames` is null. That way, most blocks do not need to have any event type list.

Reply:
Oh this is a nice idea. it could be:

```
if(getFBInterfaceSpec().mEITypeNames != nullptr) {
   return getFBInterfaceSpec().mEITypeNames[paEIID];
}
return g_nStringIdEvent;
```

And the exporter could optimize to not output event type ids when all event types are EVENT.